pages/pipelines/build_matrix.md.erb

```diff
@@ -0,0 +1,126 @@
+# Build Matrix
+
+Build Matrices lets you simplify complex build configurations by expanding a step template and array of matrix elements into multiple jobs.
+
+{:toc}
+
+For example, instead of writing three separate jobs for builds on **macOS**, **Linux** and **Windows**, like the following build configuration (which does not use a build matrix):
+
+```yaml
+steps:
+  - label: "macOS build"
+    command: "echo '.buildkite/steps/build-binary.sh macOS'"
+  - label: "Linux build"
+    command: "echo '.buildkite/steps/build-binary.sh Linux'"
+  - label: "Windows build"
+    command: "echo '.buildkite/steps/build-binary.ps1 Windows'"
+```
+{: codeblock-file="pipeline.yml"}
+
+You can use a Build Matrix, to expand a single step into three steps by replacing the `{{matrix}}` variable in the label and [command step](/docs/pipelines/command-step) with each value provided in the `matrix` array. You can interpolate the matrix values into command steps (including step labels) and into [environment variables](/docs/pipelines/environment-variables).
+
+```yaml
+steps:
+  - label: "{{matrix}} build"
+    command: "echo '.buildkite/steps/build-binary.sh {{matrix}}'"
+    matrix:
+      - "macOS"
+      - "Linux"
+      - "Windows"
+```
+{: codeblock-file="pipeline.yml"}
+
+
+
+All jobs created by a build matrix are marked with the _Matrix_ badge in the Buildkite interface.
+
+<div class="Docs__note">
+<p class="Docs__note__heading">Matrix and Parallel steps</h3>
+<p>Matrix builds are not compatible with  <a href="/docs/tutorials/parallel-builds">parallel steps</a>. You can use both in the same build, as long as they are on separate steps.</p></div>
```

Comment:
@plaindocs This note might be a little confusing, because build steps run in parallel by default in Buildkite unless you add a `wait` step. After thinking "Oh no my example for the announcement blog won't work" I realised this note is specifically about lack of support for the `parallelism` property inside of `matrix` steps. Probably worth clarifying!

Reply:
Took me a moment to wrap my head around that. So good call on clarification.